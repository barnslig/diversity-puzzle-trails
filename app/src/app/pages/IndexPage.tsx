import {
  AppBar,
  Box,
  Container,
  IconButton,
  makeStyles,
  Menu,
  MenuItem,
  Toolbar,
  Typography,
} from "@material-ui/core";
import { MoreVert } from "@material-ui/icons";
import { FormattedMessage, useIntl } from "react-intl";
import * as React from "react";

import ButtonProgressIndicator from "../../common/components/ButtonProgressIndicator";
import ChooseCharacterHeroMessage from "../../common/components/ChooseCharacterHeroMessage";
import MainNav from "../MainNav";
import Parameters from "../../features/parameters/Parameters";
import PausedHeroMessage from "../../common/components/PausedHeroMessage";
import useCharacter from "../../common/hooks/useCharacter";
import useClock from "../../common/hooks/api/useClock";
import useUpdateClock from "../../common/hooks/api/useUpdateClock";

const bg = require("./bg.svg");

const useStyles = makeStyles((theme) => ({
  root: {
    backgroundImage: `url(${bg})`,
    backgroundAttachment: "fixed",
    backgroundPositionY: theme.spacing(7), // top app bar
  },
  appBarTitle: {
    flexGrow: 1,
  },
}));

type IndexPageProps = {};

const IndexPage = (props: IndexPageProps) => {
  const classes = useStyles();
  const intl = useIntl();

  const [character] = useCharacter();

  const clock = useClock();
  const isPaused = clock.data?.data.attributes.state === "paused";

  const appBarMenuAnchorEl = React.useRef<HTMLButtonElement>(null);
  const [appBarMenuIsOpen, setAppBarMenuIsOpen] = React.useState(false);
  const [updateClockIsLoading, updateClock] = useUpdateClock();

  return (
    <div className={classes.root}>
      <AppBar position="fixed" color="inherit">
        <Toolbar>
          <Typography
            className={classes.appBarTitle}
            component="h1"
            variant="h6"
          >
            <FormattedMessage
              defaultMessage="Diversity Puzzle Trails"
              description="main page title"
            />
          </Typography>
          <IconButton
            ref={appBarMenuAnchorEl}
            aria-label={intl.formatMessage({
              defaultMessage: "Aktionen anzeigen",
              description: "appbar menu toggle",
            })}
            aria-controls="menu-appbar"
            aria-haspopup="true"
            edge="end"
            color="inherit"
            onClick={() => {
              setAppBarMenuIsOpen(!appBarMenuIsOpen);
            }}
          >
            <MoreVert />
          </IconButton>
          <Menu
            id="menu-appbar"
            anchorEl={appBarMenuAnchorEl.current}
            anchorOrigin={{
              vertical: "top",
              horizontal: "right",
            }}
            keepMounted
            transformOrigin={{
              vertical: "top",
              horizontal: "right",
            }}
            open={appBarMenuIsOpen}
            onClose={() => setAppBarMenuIsOpen(false)}
          >
            {isPaused ? (
              <MenuItem
                disabled={updateClockIsLoading}
                onClick={async () => {
                  await updateClock("running");
                  setAppBarMenuIsOpen(false);
                }}
              >
                <FormattedMessage
                  defaultMessage="Spiel für alle fortsetzen"
                  description="appbar menu item continue game"
                />
                {updateClockIsLoading && <ButtonProgressIndicator />}
              </MenuItem>
            ) : (
              <MenuItem
                disabled={updateClockIsLoading}
                onClick={async () => {
                  await updateClock("paused");
                  setAppBarMenuIsOpen(false);
                }}
              >
                <FormattedMessage
                  defaultMessage="Spiel für alle pausieren"
                  description="appbar menu item pause game"
                />
                {updateClockIsLoading && <ButtonProgressIndicator />}
              </MenuItem>
            )}
          </Menu>
        </Toolbar>
      </AppBar>
      <MainNav />

      {isPaused && <PausedHeroMessage />}

      {!character && <ChooseCharacterHeroMessage />}

      {!isPaused && character && (
        <Box component="main" paddingTop={10} paddingBottom={7}>
          <Container maxWidth="sm">
            <Parameters />
          </Container>
        </Box>
      )}
    </div>
  );
};

export default IndexPage;
