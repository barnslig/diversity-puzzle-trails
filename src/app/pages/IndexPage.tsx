import {
  AppBar,
  Box,
  Container,
  IconButton,
  makeStyles,
  Toolbar,
  Typography,
} from "@material-ui/core";
import { FormattedMessage } from "react-intl";
import { Refresh } from "@material-ui/icons";
import * as React from "react";

import MainNav from "../MainNav";
import Parameters from "../../features/parameters/Parameters";

const bg = require("./bg.svg");

const useStyles = makeStyles((theme) => ({
  root: {
    backgroundImage: `url(${bg})`,
    backgroundAttachment: "fixed",
    backgroundPositionY: theme.spacing(7), // top app bar
  },
  appBarTitle: {
    flexGrow: 1,
  },
}));

type IndexPageProps = {};

const IndexPage = (props: IndexPageProps) => {
  const classes = useStyles();

  return (
    <div className={classes.root}>
      <AppBar position="fixed" color="inherit">
        <Toolbar>
          <Typography
            className={classes.appBarTitle}
            component="h1"
            variant="h6"
          >
            <FormattedMessage
              defaultMessage="Diversity Puzzle Trails"
              description="main page title"
            />
          </Typography>
          <IconButton edge="end" color="inherit">
            <Refresh />
          </IconButton>
        </Toolbar>
      </AppBar>
      <MainNav />
      <Box component="main" paddingTop={10} paddingBottom={7}>
        <Container maxWidth="sm">
          <Parameters />
        </Container>
      </Box>
    </div>
  );
};

export default IndexPage;
