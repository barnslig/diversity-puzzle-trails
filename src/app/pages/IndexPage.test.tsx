import * as React from "react";

import renderWithIntl from "../../common/testing/renderWithIntl";

import IndexPage from "./IndexPage";

it("renders correctly", () => {
  const { container } = renderWithIntl(<IndexPage />);
  expect(container.firstChild).toMatchSnapshot();
});
