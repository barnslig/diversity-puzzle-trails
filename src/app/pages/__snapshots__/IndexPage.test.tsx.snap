// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  class="makeStyles-root-1"
>
  <header
    class="MuiPaper-root MuiAppBar-root MuiAppBar-positionFixed MuiAppBar-colorInherit mui-fixed MuiPaper-elevation4"
  >
    <div
      class="MuiToolbar-root MuiToolbar-regular MuiToolbar-gutters"
    >
      <h1
        class="MuiTypography-root makeStyles-appBarTitle-2 MuiTypography-h6"
      >
        Diversity Puzzle Trails
      </h1>
    </div>
  </header>
  <nav
    class="MuiPaper-root makeStyles-bottomNavigation-3 MuiPaper-elevation8 MuiPaper-rounded"
  >
    <div
      class="MuiBottomNavigation-root"
    >
      <button
        class="MuiButtonBase-root MuiBottomNavigationAction-root Mui-selected"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiBottomNavigationAction-wrapper"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M10 20v-6h4v6h5v-8h3L12 3 2 12h3v8z"
            />
          </svg>
          <span
            class="MuiBottomNavigationAction-label Mui-selected"
          >
            Status
          </span>
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </button>
      <button
        class="MuiButtonBase-root MuiBottomNavigationAction-root"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiBottomNavigationAction-wrapper"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M9.5 6.5v3h-3v-3h3M11 5H5v6h6V5zm-1.5 9.5v3h-3v-3h3M11 13H5v6h6v-6zm6.5-6.5v3h-3v-3h3M19 5h-6v6h6V5zm-6 8h1.5v1.5H13V13zm1.5 1.5H16V16h-1.5v-1.5zM16 13h1.5v1.5H16V13zm-3 3h1.5v1.5H13V16zm1.5 1.5H16V19h-1.5v-1.5zM16 16h1.5v1.5H16V16zm1.5-1.5H19V16h-1.5v-1.5zm0 3H19V19h-1.5v-1.5zM22 7h-2V4h-3V2h5v5zm0 15v-5h-2v3h-3v2h5zM2 22h5v-2H4v-3H2v5zM2 2v5h2V4h3V2H2z"
            />
          </svg>
          <span
            class="MuiBottomNavigationAction-label"
          >
            Scanner
          </span>
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </button>
    </div>
  </nav>
  <main
    class="MuiBox-root MuiBox-root-6 makeStyles-root-4"
  >
    <div
      class="MuiContainer-root MuiContainer-maxWidthSm"
    >
      <h2
        class="MuiTypography-root MuiTypography-h5 MuiTypography-gutterBottom"
      >
        Charakter wählen
      </h2>
      <p
        class="MuiTypography-root MuiTypography-body2 MuiTypography-colorTextSecondary"
      >
        Scanne eine Charakterkarte um am Spiel teilzunehmen. Dein gewählter Charakter gibt dir bei jeder Aktion im Spiel einen Bonus!
      </p>
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root makeStyles-arrow-5"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M20 12l-1.41-1.41L13 16.17V4h-2v12.17l-5.58-5.59L4 12l8 8 8-8z"
        />
      </svg>
    </div>
  </main>
</div>
`;
